components/server/ee/src/workspace/workspace-factory.ts

```diff
@@ -373,15 +373,22 @@ export class WorkspaceFactoryEE extends WorkspaceFactory {
             }
 
             if (OpenPrebuildContext.is(context.originalContext)) {
+                if (CommitContext.is(buildWorkspace.context)) {
+                    if (
+                        CommitContext.is(context.originalContext) &&
+                        CommitContext.computeHash(context.originalContext) !==
+                            CommitContext.computeHash(buildWorkspace.context)
```

Comment:
Also, on a side note, would you be okay with using a single `if` statement with 3 conditions, instead of two nested `if` blocks? (But I'm aware that's mostly personal preference, so please feel free to disregard this suggestion.)

Reply:
I wanted to make sure that the original logic was still available in the case that `CommitContext.is(context.originalContext)` was not true.

I wasn't sure that `originalContext` would always be a `CommitContext` and didn't want to break anything since there weren't specific unit tests for that block.

If you think it looks good, I can update it.

Reply:
> Interesting that you're using `CommitContext.computeHash` here instead of using `context.originalContext.revision` directly.

I saw it being used elsewhere when reading through some code and it made sense... I hadn't looked at the Interface for `CommitContext` too closely nor the definition for `computeHash`. Makse sense to me to use `revision` instead. Let me know what you think and I can make that change.